### PR TITLE
Debian packaging base

### PR DESCRIPTION
**Pull request by breunigs**
Hey everyone,

I guess this is a start to solve #72. 

Given the module name changed, simply renaming the Debian package and start tracking upstream doesn't seem that simple. Also, using livestreamer's most recent devel version and applying the 20 something PRs @javiercantero suggested is a lot of effort, so I got the impression helping you here is a more sensible use of my time.

Given the volatile nature of a tool such as this and the very rigid Debian process means that you'd either have to debianize "just the fixes" to the Debian packaged to version to keep it useful **or**, provide a recent version via the backports repository. Only the latter makes sense, really, for streamlink's case. So, first take away: there's no rush, if you want a working streamlink you will have to put it into backports anyway. I was asked more than once to do that for livestreamer, but declined because of lack of time.

This brings me to the next point: maybe you're better off providing it in a separate apt repository. That way you won't have to find a sponsor who will upload your package, if you decide to package on your own. And you would have to send that email every time you release a new version. There's stuff like the [openSUSE Build Service](https://en.opensuse.org/openSUSE:Build_Service) which might be of use here, but I have never used any of those.

Anyway, the patch here inherits from the livestreamer ones, which worked mostly out of the box if the renaming does not count. Of course, the `debian/changelog` would need to be cleaned and reflect the actual streamlink version `0.0.1` before it could be put to Debian at all. Similarly, you might need to update the `debian/copyright`.

The Debian packaging/building is a bit peculiar. As such, the flow I used for livestreamer more or less requires this to be maintained in a separate repository, to make it easy to build the tool in a clean environment. That is mostly useful to ensure one specified all build dependencies and the like. You can build on your Debian/Ubuntu system, using:
```
sudo apt build-dep livestreamer # or install build deps manually
git clean -fd
git reset --hard
dpkg-buildpackage -us -uc -b
```
Which will result in four packages: just the library/module for py2 and py3, the documentation and the actual program. It probably makes sense to just include everything in one packages, but I couldn't get that to work back then. Debian packaging is unfortunately not exactly straight forward. The built packages are clean:
```
lintian --pedantic --display-experimental --display-info
# I: streamlink: spelling-error-in-manpage usr/share/man/man1/streamlink.1.gz specifiying specifying
```

If you are interested, I can provide my notes on my update/build flow for livestreamer. My sponsor for new versions was/is [Piotr Ożarowski](https://people.debian.org/~piotr/sponsor), so maybe ask him to sponsor livestreamer, too, or rather, instead?

Hope that helps,
Stefan

### CODE REVIEW

**Comment by javiercantero** [comment]
> Given the volatile nature of a tool such as this and the very rigid Debian process means that you'd either have to debianize "just the fixes" to the Debian packaged to version to keep it useful or, provide a recent version via the backports repository.

Well, I don't think Debian users expect for `stable` packages to be up-to-date. I suspect that the vast majority of us that want the most recent versions have chosen to use `testing` or even `unstable` instead. I think that just maintaining streamlink reasonably up to date in `unstable` (and therefore `testing`) should be enough to keep everyone happy.

File: debian/copyright [hunk 1]
```diff
@@ -0,0 +1,72 @@
+Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Contact: Christopher Rosell <chrippa@tanuki.se>
```
Comment (bastimeyer) [inline debian/copyright]: This line still refers to the maintainer of Livestreamer and needs to be changed.

File: debian/copyright [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Source: https://github.com/streamlink/streamlink
+Upstream-Name: streamlink
+
+Files: *
+Copyright: 2011-2016 Christopher Rosell <chrippa@tanuki.se>
```
Comment (bastimeyer) [inline debian/copyright]: Another copyright line needs to be added here in addition to this one. Is 2011-2016 correct? Or is it 2011-2015?

File: debian/copyright [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+License: BSD-2-clause
+
+Files: src/streamlink/packages/pbs.py
+Copyright: 2011-2012 Andrew Moffat
+License: Expat
+
+Files: docs/_themes/sphinx_rtd_theme_violet/*
+Copyright: 2013 Dave Snider
+           2014 Christopher Rosell <chrippa@tanuki.se>
```
Comment (bastimeyer) [inline debian/copyright]: This can stay, right?

File: debian/python-streamlink-doc.doc-base [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Document: python-streamlink-doc
+Title: streamlink documentation
+Author: Christopher Rosell
```
Comment (bastimeyer) [inline debian/python-streamlink-doc.doc-base]: Another reference to the Livestreamer's maintainer.

**Comment by breunigs** [comment]
I have amended the names for copyright. Since streamlink doesn't have a shared email address, I opted to use @gravyboat's with a custom suffix. The rest is copyrighted to the team, hopefully making it future proof.

**Comment by breunigs** [comment]
@javiercantero Considering the rather tiny group of users of livestreamer ([cmp. debian popcontest statistics](https://qa.debian.org/popcon-graph.php?packages=livestreamer&show_installed=on&want_legend=on&want_ticks=on&date_fmt=%25Y-%25m&beenhere=1)) and that I have been asked a couple of times to update livestreamer on stable, I beg to differ. If I remember correctly, it was mostly people in the channel or bug reports that were saying "I have Debian and Twitch doesn't work with livestreamer". So, to some extent they expect recent versions or at least working ones.

**Comment by sbstp** [comment]
Yea I'm not sure about having this project packaged in Debian stable. This is the kind of project that needs frequent updates and can't stay frozen for 2 years. I'd rather have it in backports and sid.

**Comment by gravyboat** [comment]
@sbstp I agree, we need more regular updates than what stable can provide. Also I'm not sure if I want to be the contact point for people having issues with the package, I'll have to think about it. I wish there was a better way to do a public email that just created issues...

**Comment by fishscene** [comment]
@gravyboat I had no idea such a thing existed, but it apparently does:
https://www.google.com/#q=create+github+issue+from+e-mail

https://zapier.com/zapbook/zaps/10314/create-github-issues-from-new-emails-on-gmail/

**Comment by gravyboat** [comment]
@fishscene Yeah I saw that actually. I was looking for something where I didn't have to give a third party app email access. I wish there was a GitHub endpoint for it.

**Comment by gravyboat** [comment]
I'll just bite the bullet and we can use my email for now.

**Comment by javiercantero** [comment]
Note that Debian Stretch has entered the soft freeze [(announce)](https://lists.debian.org/debian-devel-announce/2017/01/msg00002.html), so it's no longer possible to ship streamlink in it (if anyone wanted to).

**Comment by gravyboat** [comment]
@javiercantero Thanks for the heads up, I don't think we had any luck talking to the existing sponsor (I think someone approached them but maybe it was a different distro).

**Comment by stepshal** [comment]
@sbstp @gravyboat backports is just packages from testing so it's gonna end up in stable anyway.

**Pull request merged**